src/datadog/config_manager.h

```diff
@@ -10,54 +12,32 @@
 namespace datadog {
 namespace tracing {
 
-struct ConfigUpdate {
-  Optional<TraceSamplerConfig> trace_sampler;
-};
-
 class ConfigManager {
   mutable std::mutex mutex_;
   Clock clock_;
   std::shared_ptr<TraceSampler> default_trace_sampler_;
   std::shared_ptr<TraceSampler> current_trace_sampler_;
 
  public:
-  ConfigManager(const FinalizedTracerConfig& config)
-      : clock_(config.clock),
-        default_trace_sampler_(
-            std::make_shared<TraceSampler>(config.trace_sampler, clock_)),
-        current_trace_sampler_(default_trace_sampler_) {}
+  struct Update {
+    Optional<TraceSamplerConfig> trace_sampler;
+  };
```

Comment:
I saw three options:

1. Leave it as you had it.
2. Put `struct ConfigUpdate` in its own component (its own `.h`). Good place for documentation!
3. Consider `ConfigUpdate` as part of `ConfigManager`'s public interface.

(1) and (3) are compatible, but so too are (3) and my proposed change.

My reasoning was "if I see `ConfigUpdate` used in `RemoteConfigManager`, where do I go looking for its definition?" I might look for a `config_update.h`, but there is none. `ConfigUpdate` is a parameter to `ConfigManager`, so I might look there, but the connection is even stronger if it's spelled `ConfigManager::Update`.

> It can also be interpreted as an `Update` of the `ConfigManager`, which is not the intended meaning.

That's a good point. I prefer (2) to (1) if you're looking for a compromise. I'm also fine with (1). This PR is illustrative only.